compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeclarationStubGenerator.kt

```diff
@@ -120,6 +120,7 @@ class DeclarationStubGenerator(
                     constantValueGenerator.generateConstantValueAsExpression(UNDEFINED_OFFSET, UNDEFINED_OFFSET, it)
                 )
             }
+            parent = generateParentStub(descriptor)
```

Comment:
Right, that was my intent. Previously deserialized IrFields ended up with unset parents. Is there a reason why this would be the desired behavior?

Reply:
`parent` should be generated only for synthetic declarations because everything comes from deserializer is properly initialized. If you have an example when it is not true could you please provide it?

Reply:
Seems this piece of original code is totally wrong, there should be IrLazyFieldImpl with lazy parent and initializer calculation

Reply:
Similar to another 'declares', e.g.:
```
return symbolTable.declareClass(UNDEFINED_OFFSET, UNDEFINED_OFFSET, origin, descriptor) {
            deserializer?.findDeserializedDeclaration(referenceClass) as? IrClass
                ?: IrLazyClass(UNDEFINED_OFFSET, UNDEFINED_OFFSET, origin, it, this, typeTranslator)
        }
```

Reply:
@sfs This part is fixed in my branch: 
https://github.com/JetBrains/kotlin/commits/rr/mb/lazyfield

Reply:
And pushed to master

Reply:
Great, thank you! I got rid of this part in the PR.